libsolidity/formal/SMTEncoder.cpp

```diff
@@ -107,6 +107,9 @@ void SMTEncoder::endVisit(ContractDefinition const& _contract)
 
 	solAssert(m_currentContract == &_contract, "");
 	m_currentContract = nullptr;
+
+	if (m_callStack.empty())
+		m_context.popSolver();
```

Comment:
`initContract` pushes (resp. `initFunction`), so this should pop the same way `endVisit(FunctionDefinition)` does